core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -523,16 +524,32 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
 
     val exec = content.exec getOrElse action.exec
 
-    WhiskAction(
-      action.namespace,
-      action.name,
-      exec,
-      parameters,
-      limits,
-      content.version getOrElse action.version.upPatch,
-      content.publish getOrElse action.publish,
-      (content.annotations getOrElse action.annotations) ++ execAnnotation(exec))
-      .revision[WhiskAction](action.docinfo.rev)
+    if (content.unlock.getOrElse(false)) {
+      WhiskAction(
+        action.namespace,
+        action.name,
+        exec,
+        parameters,
+        limits,
+        content.version getOrElse action.version.upPatch,
+        content.publish getOrElse action.publish,
+        (content.annotations getOrElse action.annotations) ++ execAnnotation(exec) ++ Parameters(
+          WhiskAction.lockFieldName,
+          JsBoolean(false)))
+        .revision[WhiskAction](action.docinfo.rev)
+    } else {
+      //keep original value not changed
+      WhiskAction(
+        action.namespace,
+        action.name,
+        exec,
+        parameters,
+        limits,
+        content.version getOrElse action.version.upPatch,
+        content.publish getOrElse action.publish,
+        (content.annotations getOrElse action.annotations) ++ execAnnotation(exec))
```

Comment:
Instead of having the if statement and duplicate code, can you just keep the original `WhiskAction` block and add something like the below to the annotations?

```
++ content.unlock.map(u => Parameters(WhiskAction.lockFieldName, JsBoolean(false))).getOrElse(Parameters())
```

Reply:
@dubee ,thanks, already fixed.